### PR TITLE
Sync WP Version

### PR DESCRIPTION
**Pull request by enejb**
Let us sync the WP version separately. 
This PR synced the WP version on every heartbeat as well as after the user updates WordPress Core.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -310,6 +310,10 @@ public static function init() {
 	 * Must never be called statically
 	 */
 	function plugin_upgrade() {
+		if( isset( $_GET['sync'] ) ) {
+			error_log('sync do action called ' );
+			do_action( 'jetpack_sync_all_registered_options' );
```
Comment (jeherve) [inline class.jetpack.php]: Could you add the following before the action?

`/** This action is documented in class.jetpack.php */`

File: class.jetpack.php [hunk 2]
```diff
@@ -310,6 +310,10 @@ public static function init() {
 	 * Must never be called statically
 	 */
 	function plugin_upgrade() {
+		if( isset( $_GET['sync'] ) ) {
+			error_log('sync do action called ' );
```
Comment (jeherve) [inline class.jetpack.php]: You'll probably want to remove that `error_log()`. I assume it was there for debugging?

File: class.jetpack.php [hunk 3]
```diff
@@ -988,6 +1002,20 @@ public static function is_single_user_site() {
...
+	 * Sync back wp_version
+	 */
+	public static function get_wp_version() {
+		global $wp_version;
+		return $wp_version;
+	}
+
+	public static function update_get_wp_version( $update, $meta_data ) {
+		if ( 'update' === $meta_data['action'] && 'core' === $meta_data['type'] ) {
+			do_action( 'add_option_jetpack_wp_version', 'jetpack_wp_version', (string) (bool) Jetpack::get_wp_version() );
```
Comment (jeherve) [inline class.jetpack.php]: Could you document that action?

File: class.jetpack.php [hunk 4]
```diff
@@ -988,6 +998,24 @@ public static function is_single_user_site() {
...
+		return $wp_version;
+	}
+	/**
+	 * Keeps wp_version in sync with .com when WordPress core updates 
+	 **/
+	public static function update_get_wp_version( $update, $meta_data ) {
+		if ( 'update' === $meta_data['action'] && 'core' === $meta_data['type'] ) {
+			/** This action is documented in wp-includes/option.php */
+			/* This triggers the sync for the jetpack version */
+			do_action( 'add_option_jetpack_wp_version', 'jetpack_wp_version', (string) (bool) Jetpack::get_wp_version() );
```
Comment (zinigor) [inline class.jetpack.php]: I'm pretty sure you didn't mean to use both `(bool)` and `(string)` here, because otherwise we'll always get `1` as a result.

Comment (samhotchkiss) [inline class.jetpack.php]: Yeah, this shouldn't ever be bool.  Also, needs a rebase

**Comment by enejb** [comment]
Updated the comment to be more helpful as well as fixed the code with suggestions.

**Comment by zinigor** [comment]
Thank you! Looks great!

**Pull request merged**